src/bitcoin-cli.cpp

```diff
@@ -191,7 +192,9 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
 
 UniValue CallRPC(const std::string& strMethod, const UniValue& params)
 {
-    std::string host = GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
+    std::string host;
+    int dummy_port;
+    SplitHostPort(GetArg("-rpcconnect", DEFAULT_RPCCONNECT), dummy_port, host);
```

Comment:
sure - changed to use the port from `-rpcconnect` if it's specified.

It was a bit more than a one line change because it was a little fiddly to get the precedence correct:

1. -rpcport if there is one; else
2. port in -rpcconnect if there is one; else
3. default port for the chain